File: api/airTicket/src/main/java/com/xiang/airTicket/entity/AirlineCompany.java
package com.xiang.airTicket.entity;

import javax.persistence.*;

// 航空公司
@Entity
public class AirlineCompany {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // id

    private String name; // 名称

    private String icon; // 图标

    @ManyToOne
    private City city; // 所属城市


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }
}

File: webapp/airTicket/src/app/entity/AirlineCompany.ts
// 航空公司

import {City} from './City';

export class AirlineCompany {

  id: number; // id

  name: string; // 名称

  icon: string; // 图标

  city: City; // 所属城市

  constructor(id: number, name: string, icon: string, city: City) {
    this.id = id;
    this.name = name;
    this.icon = icon;
    this.city = city;
  }
}
